### ISSUE TITLE
Support separate git directories

### ISSUE THREAD

**Issue by amw** [git-up/GitUp#745]
Git has a feature where tracked files and the git metadata are kept in separate directories. This is useful for example if you want to track changes to cloud-synced directory without syncing git objects themselves.

```
git init --separate-git-dir=../Integrity/Documents
```

From git help
>        --separate-git-dir=<git dir>
>           Instead of initializing the repository as a directory to either $GIT_DIR or ./.git/, create
>           a text file there containing the path to the actual repository. This file acts as
>           filesystem-agnostic Git symbolic link to the repository.

In my example the `.git` text file contains only this:
```
gitdir: /Users/amw/Integrity/Documents
```

When I try to open `/Users/amw/Documents` in GitUp I receive the following error message:
> could not find repository from '/Users/amw/Integrity'

**Comment by amw** [comment]
Actually, I was just able to open up this directories through File -> Open dialog. I only got the error when I was using `gitup .` shell command. I can't reproduce the error anymore, but `gitup .` still does not open the tab/window if it doesn't already exist.

You can close the issue if you want to – unless you think there is something that can be done about improving reliability of this.

**Comment by lucasderraugh** [comment]
Does it always fail from the command line? If so then we can change the title of the ticket and leave it open. Ideally the command line should open anything that the GUI open dialog can.

**Comment by amw** [comment]
Yeah, I can never open repo from the command line. It used to show an error, now it doesn't, but still it doesn't add the tab.

**Comment by char5280** [comment]
> Git has a feature where tracked files and the git metadata are kept in separate directories. This is useful for example if you want to track changes to cloud-synced directory without syncing git objects themselves.
> 
> ```
> git init --separate-git-dir=../Integrity/Documents
> ```
> 
> From git help
> >        --separate-git-dir=<git dir>
> >           Instead of initializing the repository as a directory to either $GIT_DIR or ./.git/, create
> >           a text file there containing the path to the actual repository. This file acts as
> >           filesystem-agnostic Git symbolic link to the repository.
> 
> In my example the `.git` text file contains only this:
> ```
> gitdir: /Users/amw/Integrity/Documents
> ```
> 
> When I try to open `/Users/amw/Documents` in GitUp I receive the following error message:
> > could not find repository from '/Users/amw/Integrity'

**Comment by julasamer** [comment]
I have a similar (or the same) issue. When I use gitup from the command line, it says "could not find repository from [...]/.repo/projects/[...]". The .git directory in the project is an alias to the directory mentioned in the error message.

This is because we use repo, which stores the .git directories in it's .repo directory.

Opening the same project from Gitup's open menu works fine.
